// ng-tournament-tree/esm5/lib/single-elimination-tree/single-elimination-tree.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input, TemplateRef } from '@angular/core';
var SingleEliminationTreeComponent = /** @class */ (function () {
    function SingleEliminationTreeComponent() {
    }
    /**
     * @return {?}
     */
    SingleEliminationTreeComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
    };
    /**
     * @param {?} changes
     * @return {?}
     */
    SingleEliminationTreeComponent.prototype.ngOnChanges = /**
     * @param {?} changes
     * @return {?}
     */
    function (changes) {
        if (changes.hasOwnProperty('tournament') && changes.tournament.currentValue) {
            this.winnersBracket = this.tournament.rounds.filter((/**
             * @param {?} round
             * @return {?}
             */
            function (round) {
                return round.type === 'Winnerbracket';
            }));
        }
        console.log(this.winnersBracket);
        this.final = this.tournament.rounds.filter((/**
         * @param {?} round
         * @return {?}
         */
        function (round) {
            return round.type === 'Final';
        })).shift();
        console.log(this.final);
    };
    SingleEliminationTreeComponent.decorators = [
        { type: Component, args: [{
                    selector: 'ngtt-single-elimination-tree',
                    template: "<div class=\"ngtt-se__container\">\r\n  <div class=\"ngtt-se__bracket-wrapper\">\r\n    <ng-container *ngFor=\"let winnersround of winnersBracket; let i = index\">\r\n      <div class=\"ngtt-se__round-wrapper\">\r\n        <ng-container *ngFor=\"let match of winnersround.matches\">\r\n          <div class=\"ngtt-se__match-wrapper\">\r\n            <div class=\"ngtt-se__match-component-container\">\r\n              <ng-container *ngTemplateOutlet=\"matchTemplate; context:{$implicit: match}\"></ng-container>\r\n            </div>\r\n          </div>\r\n        </ng-container>\r\n      </div>\r\n    </ng-container>\r\n    <ng-container>\r\n      <div class=\"ngtt-se__round-wrapper\">\r\n        <div class=\"ngtt-se__match-wrapper\">\r\n          <div class=\"ngtt-se__finale-match-wrapper\">\r\n            <div class=\"ngtt-se__finale-component-container\">\r\n              <ng-container\r\n                *ngTemplateOutlet=\"matchTemplate; context:{$implicit: final.matches[0]}\"></ng-container>\r\n            </div>\r\n            <div class=\"ngtt-se__match-component-container\" *ngIf=\"final.matches.length > 1\">\r\n              <ng-container *ngTemplateOutlet=\"matchTemplate; context:{$implicit: final.matches[1]}\"></ng-container>\r\n            </div>\r\n          </div>\r\n        </div>\r\n      </div>\r\n    </ng-container>\r\n  </div>\r\n</div>\r\n",
                    styles: [""]
                }] }
    ];
    /** @nocollapse */
    SingleEliminationTreeComponent.ctorParameters = function () { return []; };
    SingleEliminationTreeComponent.propDecorators = {
        matchTemplate: [{ type: Input }],
        tournament: [{ type: Input }]
    };
    return SingleEliminationTreeComponent;
}());
export { SingleEliminationTreeComponent };
if (false) {
    /** @type {?} */
    SingleEliminationTreeComponent.prototype.matchTemplate;
    /** @type {?} */
    SingleEliminationTreeComponent.prototype.tournament;
    /** @type {?} */
    SingleEliminationTreeComponent.prototype.winnersBracket;
    /** @type {?} */
    SingleEliminationTreeComponent.prototype.final;
}
